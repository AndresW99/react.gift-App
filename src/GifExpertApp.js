import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ( { titulo } ) => {

    const [categories, setCategories] = useState(['']);

    return (
        <>
            <h2 className="Gifs">Gifs finder App with:</h2>
            <h2 className="titulo"> { titulo }</h2> 
            <h3 className="Sub">Developed and designed by: Andres Wellmann.</h3>

            <hr />
           
            <AddCategory setCategories={ setCategories }/>
           
            <hr /> 

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                        key={ category }
                        category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}

GifExpertApp.propTypes = {
    titulo: PropTypes.string
}



export default GifExpertApp;