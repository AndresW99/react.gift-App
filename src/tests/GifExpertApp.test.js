import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el GifExpertApp', () => {

    const wrapper = shallow( <GifExpertApp />)

    test('Tomar un snapshot del codigo.', () => {
        expect( wrapper ).toMatchSnapshot();     
    });
    
});
