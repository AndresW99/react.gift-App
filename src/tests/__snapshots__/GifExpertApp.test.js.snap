// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp Tomar un snapshot del codigo. 1`] = `
<Fragment>
  <h2
    className="Gifs"
  >
    Gifs frinder App with:
  </h2>
  <h2
    className="titulo"
  >
     
  </h2>
  <h3
    className="Sub"
  >
    Developed and designed by: Andres Wellmann.
  </h3>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Overwatch"
      key="Overwatch"
    />
  </ol>
</Fragment>
`;
